pootle/apps/virtualfolder/models.py

```diff
@@ -355,6 +366,13 @@ def save(self, *args, **kwargs):
             self.directory.pootle_path
         )
 
+        # Abort if there is some clash with an existing directory.
+        if Directory.objects.filter(pootle_path=self.pootle_path).exists():
```

Comment:
This is pure field validation, and it's better off in `clean_fields()` as `VirtualFolder`s do. Also, no apparent need for custom exceptions, this is just `ValidationError`.